From 451ad0b567b1a04a859b9ca64d56f029730708b5 Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Tue, 17 May 2022 12:58:37 +0100
Subject: [PATCH] media: imx296: Fix Bayer order

Switch the Bayer ordering advertised by the device driver from BGGR to RGGB.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/i2c/imx296.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx296.c b/drivers/media/i2c/imx296.c
index 4be03180deac..33944155a6ce 100644
--- a/drivers/media/i2c/imx296.c
+++ b/drivers/media/i2c/imx296.c
@@ -610,7 +610,7 @@ static int imx296_enum_mbus_code(struct v4l2_subdev *sd,
 		return -EINVAL;
 
 	code->code = sensor->mono ? MEDIA_BUS_FMT_Y10_1X10
-		   : MEDIA_BUS_FMT_SBGGR10_1X10;
+		   : MEDIA_BUS_FMT_SRGGB10_1X10;
 
 	return 0;
 }
@@ -715,7 +715,7 @@ static int imx296_set_format(struct v4l2_subdev *sd,
 	}
 
 	format->code = sensor->mono ? MEDIA_BUS_FMT_Y10_1X10
-		     : MEDIA_BUS_FMT_SBGGR10_1X10;
+		     : MEDIA_BUS_FMT_SRGGB10_1X10;
 	format->field = V4L2_FIELD_NONE;
 	format->colorspace = V4L2_COLORSPACE_RAW;
 	format->ycbcr_enc = V4L2_YCBCR_ENC_DEFAULT;
-- 
2.32.0

